sdk/storage/Azure.Storage.Files.Shares/src/ShareExtensions.cs

```diff
@@ -39,6 +40,21 @@ internal static string ToFileDateTimeString(this DateTimeOffset? dateTimeOffset)
         private static string ToFileDateTimeString(this DateTimeOffset dateTimeOffset)
             => dateTimeOffset.UtcDateTime.ToString(Constants.File.FileTimeFormat, CultureInfo.InvariantCulture);
 
+        internal static string ToShareEnableProtocolsString(this ShareEnabledProtocols? shareEnabledProtocols)
+        {
+            if (shareEnabledProtocols == null)
+            {
+                return null;
+            }
+
+            return shareEnabledProtocols switch
+                {
+                    ShareEnabledProtocols.Smb => Constants.File.SmbProtocol,
+                    ShareEnabledProtocols.Nfs => Constants.File.NfsProtocol,
+                    _ => throw new ArgumentException($"Unknown share protocol: {shareEnabledProtocols}"),
+                };
```

Comment:
should this concat? we know we'll have that coming in the future

Reply:
I think it is fine for now, I will update it in the future.

Reply:
What happens if someone provides `ShareEnabledProtocols.Smb | ShareEnabledProtocols.Nfs` ?

Reply:
I think will throw an exception, it is not current valid to provide both.

Reply:
Let me double check.

Reply:
yeah. If we provide flags enum I think SDK shouldn't be doing this validation and defer to service to throw.

Reply:
If we don't throw the Exception, we will accept whichever protocol the user specified first, and create the share with it.  This is not the intended behavior for the user, to silently modified what they requested and succeed.  I think we should leave this as-is.

Reply:
Note that I did have to change the flags enum to use 1 and 2, so this validation actually works.  Previously, it was not.